### DISCUSSION TITLE
Can't generate hover variant for padding

### DISCUSSION THREAD

**Discussion by ngocquyhoang** [tailwindlabs/tailwindcss#2876] [Help] [answered]
I'm trying to generate hover variant for `padding`, I follow [this](https://tailwindcss.com/docs/hover-focus-and-other-states#hover) document but it doesn't seem to work.
This is my `tailwind.config.js`

```
module.exports = {
	future: {
		removeDeprecatedGapUtilities: true,
		purgeLayersByDefault: true
	},
	purge: {
		enabled: false,
		content: ['./public/*.html', './src/**/*.vue']
	},
	theme: {
		extend: {
			backgroundImage: {
				sunset: "url('../../images/sunset.jpg')"
			},
			backgroundSize: {
				'32': '8rem',
				'64': '16rem'
			},
			keyframes: {
				zoomIn: {
					'0%': { transform: 'scale(0)' },
					'100%': { transform: 'scale(1)' }
				}
			},
			animation: {
				'zoom-in-fast': 'zoomIn 0.15s ease-in-out'
			},
			colors: {
				'primary-100': '#D6E4FF'
			}
		}
	},
	variants: {
		extend: {
			padding: ['hover']
		}
	},
	plugins: []
};

```

**Comment by MathGeniusJodie** [comment]
if I paste your config in https://play.tailwindcss.com/ the classes get generated fine, I think purgecss is removing them

**Reply by ngocquyhoang** [reply]
Thank you for your time helping me.
I think the problem is not in the purecss, because even when I remove purecss it still doesn't work.
When I try  [https://play.tailwindcss.com/](https://play.tailwindcss.com/), I can see it. 
<img width="360" alt="Screen Shot 2020-11-26 at 0 07 28" src="https://user-images.githubusercontent.com/10533371/100245863-da9ae800-2f7b-11eb-95d4-55771001d6df.png">

<img width="334" alt="Screen Shot 2020-11-26 at 0 07 51" src="https://user-images.githubusercontent.com/10533371/100245889-e1295f80-2f7b-11eb-8072-05303439d2f0.png">

Go back to my code,  I tried to drop `extend`, and unexpectedly it worked. 
I am not sure if I remove extend will I have problems with the other components?

**Comment by simonswiss** [comment, marked as answer]
Hello! 👋

What version of Tailwind CSS are you using?

The way you extend the `padding` plugin with `variants.extend.padding` is correct.

Here's a Tailwind Play showing it working on hover: 

https://play.tailwindcss.com/aWFzGUc7rH?file=config

**Reply by ngocquyhoang** [reply]
I'm sorry for forgetting to mention the version. Currently I am using tailwind 1.9.6

**Comment by ngocquyhoang** [comment]
After I updated to 2.0.1 (from 1.9.6) and added `extend` back, it work like a god. 
Thank you very much @MathGeniusJodie @simonswiss

**Comment by ngocquyhoang** [comment]
After I updated to 2.0.1 (from 1.9.6) and added extend back, it worked like a god. 
Thank you very much @MathGeniusJodie @simonswiss
